projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/floating/LLVM80BitFloatArithmeticNode.java

```diff
@@ -29,51 +29,90 @@
  */
 package com.oracle.truffle.llvm.nodes.op.arith.floating;
 
+import com.oracle.truffle.api.dsl.Cached;
 import com.oracle.truffle.api.dsl.NodeChild;
 import com.oracle.truffle.api.dsl.NodeChildren;
 import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropException;
 import com.oracle.truffle.llvm.runtime.floating.LLVM80BitFloat;
+import com.oracle.truffle.llvm.runtime.nodes.api.LLVMArithmetic.LLVMArithmeticOpNode;
 import com.oracle.truffle.llvm.runtime.nodes.api.LLVMExpressionNode;
 
 @NodeChildren({@NodeChild("leftNode"), @NodeChild("rightNode")})
 public abstract class LLVM80BitFloatArithmeticNode extends LLVMExpressionNode {
+    protected LLVMArithmeticOpNode createAddNode() {
+        return LLVMArithmeticFactory.createAddNode();
+    }
 
-    public abstract static class LLVM80BitFloatAddNode extends LLVM80BitFloatArithmeticNode {
-        @Specialization
-        protected LLVM80BitFloat add(LLVM80BitFloat left, LLVM80BitFloat right) {
-            return left.add(right);
-        }
+    protected LLVMArithmeticOpNode createSubNode() {
+        return LLVMArithmeticFactory.createSubNode();
     }
 
-    public abstract static class LLVM80BitFloatMulNode extends LLVM80BitFloatArithmeticNode {
+    protected LLVMArithmeticOpNode createMulNode() {
+        return LLVMArithmeticFactory.createMulNode();
+    }
+
+    protected LLVMArithmeticOpNode createDivNode() {
+        return LLVMArithmeticFactory.createDivNode();
+    }
 
+    protected LLVMArithmeticOpNode createRemNode() {
+        return LLVMArithmeticFactory.createRemNode();
+    }
+
+    public abstract static class LLVM80BitFloatAddNode extends LLVM80BitFloatArithmeticNode {
         @Specialization
-        public LLVM80BitFloat execute80BitFloat(LLVM80BitFloat left, LLVM80BitFloat right) {
-            return left.mul(right);
+        protected LLVM80BitFloat add(VirtualFrame frame, LLVM80BitFloat left, LLVM80BitFloat right, @Cached("createAddNode()") LLVMArithmeticOpNode node) {
+            try {
+                return (LLVM80BitFloat) node.execute(frame, left, right);
+            } catch (InteropException e) {
+                throw e.raise();
```

Comment:
I think there is a `CompilerDirectives.transferToInterpreter();` missing here. Same for the other specializations in this file and the `AssertionError` occurances in the `createNode` methods of `LLVMArithmeticFactory`.